providers/email_setup.rb

```diff
@@ -250,7 +252,8 @@ def setup_gecos_as_default_profile(profile_ini)
           plugin_file = "/var/cache/gecos/email_setup/#{plugin_file_name}"
 
           # Download extension if necessary
-          remote_file plugin_file do
+          remote_file '#{plugin_file}-#{username}' do
```

Comment:
I think extensions must be identified as unique resources using "plugin-file", not "plugin-file" and "username"

Reply:
Presently plugins are installed inside a users loop (starting in line 85), thus "username" is required in order to have unique resource names. I'll check if plugins download can be moved outside the users loop.

Reply:
Plugins to be installed are included in email templates, there can be n different templates in the databag, the template to be used is especified per user. Conclusion: plugin download can not be taken out of the users loop.

Reply:
Ok, let's merge and test all these changes